components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -3443,6 +3454,8 @@ public IdentityProvider getIdPByAuthenticatorPropertyValue(Connection dbConnecti
                 if (defaultAuthenticatorName != null) {
                     FederatedAuthenticatorConfig defaultAuthenticator = new FederatedAuthenticatorConfig();
                     defaultAuthenticator.setName(defaultAuthenticatorName);
+                    defaultAuthenticator.setDefinedByType(DefinedByType.valueOf(
+                            defaultAuthenticatorDefinedByType));
```

Comment:
Can we consider the `Defined_By=SYSTEM`, if a value is not set in the database?

Reply:
As Defined_By is not null column in the database, there cannot be a scenario to that a value is not being set in the database.